libbeat/common/config.go

```diff
@@ -263,10 +263,16 @@ func (c *Config) PrintDebugf(msg string, params ...interface{}) {
 	}
 }
 
+// Enabled return the configured enabled value or true by default.
 func (c *Config) Enabled() bool {
+	return c.EnabledWithDefault(true)
+}
+
+// EnabledWithDefault returns the configured value or the default value if the enabled is not set.
+func (c *Config) EnabledWithDefault(v bool) bool {
```

Comment:
We used to have some enabled settings being `true` and others being `false` by default in the past. So to reduce confusion we did decide to have `enabled` to always default to `true`, asking users to explicitely adding `enabled: false` if required (cheap alternative to commenting out setting blocks).

This is default behavior for 'everything' is reflected by 'Enabled()'. Do we really need/want to re-introduce `enabled: false` by default? (Unfortunately ILM support will become an exception introducing 'auto').

Reply:
is is indeed a bit hacky let me come up with a better thing.

Reply:
Thanks, I didn't know the back story I will remove that.